go/fn/object.go

```diff
@@ -169,20 +169,41 @@ func (o *SubObject) NestedSliceOrDie(fields ...string) SliceSubObjects {
 	return val
 }
 
-// NestedInt64 returns the int64 value, if the field exist and a potential error.
-func (o *SubObject) NestedMap(fields ...string) (*SubObject, bool, error) {
-	var variant internal.MapVariant
+// NestedMap returns a map[string]string value of a nested field, false if not found and an error if not a map[string]string type.
```

Comment:
godoc out-of-date.

Reply:
Same as https://github.com/GoogleContainerTools/kpt-functions-sdk/pull/573#discussion_r875107750, will do that in a follow up PR since these two have conflicts.